app/services/classification/base.rb

```diff
@@ -133,6 +137,18 @@ def path_elements(path = nil)
 
     private
 
+    def check_valid(key, value)
+      lookup_value = key.lookup_values.build(:value => value)
+      raise "Invalid value '#{value}' of parameter #{key.id} '#{key.key}'" unless lookup_value.send(:validate_list)
```

Comment:
I think this would be a bit nicer returning early if everything is okay, otherwise it's one extra string we have to maintain for no special reason (they are the same).

```ruby
return if lookup_value.send(:validate_list) && lookup_value.send(:validate_regexp)
raise "Invalid value '#{value}' of parameter #{key.id} '#{key.key}'"
```